src/hooks/useDateForm.test.tsx

```diff
@@ -0,0 +1,116 @@
+import {
+  endDateInvalid,
+  rangeContainsInvalidYearFormat,
+  useDateForm,
+} from "./useDateForm"
+import { renderHook, screen, render } from "../utils/testUtils"
+import type { MutableRefObject } from "react"
+import { useRef } from "react"
+import type { TextInputRefType } from "@nypl/design-system-react-components"
+import DateForm from "../components/SearchFilters/DateForm"
+
+const setUpDateFormWithRefs = (start, end) => {
+  let refs: MutableRefObject<TextInputRefType>[]
+  // this component exists only to populate the ref values
+  const TestComponent = () => {
+    refs = [useRef<TextInputRefType>()]
+    return null
+  }
+  render(<TestComponent />)
+  const {
+    rerender,
+    result: {
+      current: { validateDateRange, dateFormProps },
+    },
+  } = renderHook(() =>
+    useDateForm({
+      inputRefs: refs,
+      dateAfter: start,
+      dateBefore: end,
+      changeHandler: () => true,
+    })
+  )
+  return { validateDateRange, dateFormProps, rerender }
+}
+
+describe("useDateForm", () => {
+  describe("validateDateRange", () => {
+    it("validateDateRange focuses the input when end date is earlier than start date", () => {
+      const { validateDateRange, dateFormProps } = setUpDateFormWithRefs(
+        "2000",
+        "1999"
+      )
+      render(<DateForm {...dateFormProps} />)
+      validateDateRange()
+      const input1 = screen.getByDisplayValue("2000")
+      const input2 = screen.getByDisplayValue("1999")
+      expect(input2).not.toHaveFocus()
+      expect(input1).toHaveFocus()
+    })
+    it("validateDateRange returns false when end date is earlier than start date", async () => {
+      const { validateDateRange, dateFormProps } = setUpDateFormWithRefs(
+        "2000",
+        "1999"
+      )
+      render(<DateForm {...dateFormProps} />)
+      expect(validateDateRange()).toBe(false)
+    })
+    it.todo(
```

Comment:
The hook works but this is hard to test. I tried this and the asynchronous nature of `setDisplayDateRangeError` is not updating the value for `displayDateRangeError` right away. That means that the returned `dateFormWithHookProps` object does not have the latest error message. I was testing to see if i could call `setUpDateFormWithRefs` to get the "updated" value for `dateFormProps` but this doesn't work and isn't correct.

The way the hook works, it sort of depends on the `DateForm` component to rerender and get the updated value. I think you can ignore/remove this test and instead add a test to verify the error message on the screen for the `DateForm` component.